app/src/main/java/org/fossasia/openevent/general/event/EventDao.kt

```diff
@@ -2,13 +2,15 @@ package org.fossasia.openevent.general.event
 
 import android.arch.persistence.room.Dao
 import android.arch.persistence.room.Insert
+import android.arch.persistence.room.OnConflictStrategy.IGNORE
 import android.arch.persistence.room.OnConflictStrategy.REPLACE
 import android.arch.persistence.room.Query
 import io.reactivex.Flowable
+import io.reactivex.Single
 
 @Dao
 interface EventDao {
-    @Insert(onConflict = REPLACE)
+    @Insert(onConflict = IGNORE)
```

Comment:
Why ignore?

Reply:
because the entries are overwritten every time events are loaded from network thereby losing all favourite related data

Reply:
So, there must be a better solution. This will remove the opportunity for updated event info

Reply:
nikit is working on swipe refresh layout we can add force update there

Reply:
Why break functionality to fix it later?

Reply:
okay so I will revert it back to replace and then we can change it in swipe refresh PR

Reply:
But this is not related to swipe refresh?

This is related to favorites

Reply:
one thing we can do is that before inserting into db we can select ids of all favourite events from db and after inserting set those events favourite again. How does this sound to you?

Reply:
Yes, that's exactly what I meant :)

Reply:
```
 return eventDao.getFavoriteEvents().singleOrError().map {
            it.map { it.id }
        }.flatMap { favEvents ->
            eventApi.searchEvents("location", locationName).map {
                eventDao.insertEvents(it)
                eventDao.setFavorites(favEvents)
                it
            }.flatMap {
                eventDao.getEventsWithIds(it.map { it.id }.toList())
            }
        }
```
@iamareebjamal doing it this way would be right?

Reply:
This still has problems with single queries running multiple times, test it

Reply:
how do i check, i cant put log statements at DAO level? and this method is also not working 😞 .

Reply:
First of all, ditch the it.map wherever you are using, it's non standard. Always use chaining operators.

This should be your logic:

- [ ] Take the IDs of events being saved into the database
- [ ] SELECT id, favorite from Event where id in :eventIds
- [ ] This will give you list of objects with id and favorite
- [ ] Set the old favorites on new event objects by matching the ID
- [ ] Save them

Reply:
almost done just one quick doubt
```
  fun function(apiEvents: List<Event>, favEventIds: List<Long>): List<Event> {
            eventDao.insertEvents(apiEvents)
            return apiEvents
        }

        return Single.zip(
                eventApi.searchEvents("name", locationName),
                eventDao.getFavouriteEventIds(),
                BiFunction<List<Event>, List<Long>, List<Event>> { t1, t2 -> function(t1, t2) }
        )
```
since these objects `apiEvents` and `favEventIds` are immutable how do i change the field of `apiEvent` events objects before inserting

Reply:
make favorite in event model mutable